src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -108,4 +111,25 @@ private static CredentialsMatcher githubScanCredentialsMatcher() {
 
     private Connector() {}
 
+    /**
+     * Fail immediately and throw a customized exception.
+     */
+    public static final RateLimitHandler CUSTOMIZED = new RateLimitHandler() {
```

Comment:
Should not be this used for all connections? (missed in [line 77](https://github.com/jenkinsci/github-branch-source-plugin/pull/11/files#diff-61948c4cd4e480f18c6e572cf88a055bR77)) Probably nobody reaches the rate limit being authenticated, but who knows...